Bottom: 8c4c6886c2d578ba7dbfc95c2e3c165f9cffdf76
Top:    780c38ba10b69e8c661b706bca3411ca87b8f583
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-05-13 02:26:52 +0100

gtk/gl: Don't set "magenta" for a NULL color passed to ghid_set_color

The code has a strdup (name), followed by a test for name == NULL. If
name _were_ ever NULL, the strdup would crash, so we can remove this
test.


---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index ad5ff98..3902bc3 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -369,14 +369,6 @@ ghid_set_color (hidGC gc, const char *name)
     }
 
   old_name = strdup (name);
-
-  if (name == NULL)
-    {
-      fprintf (stderr, "%s():  name = NULL, setting to magenta\n",
-               __FUNCTION__);
-      name = "magenta";
-    }
-
   gc->colorname = (char *) name;
 
   if (gport->colormap == NULL)
